pkg/network/ebpf/c/prebuilt/http.c

```diff
@@ -52,6 +67,177 @@ int kretprobe__tcp_sendmsg(struct pt_regs* ctx) {
     return 0;
 }
 
+static __always_inline conn_tuple_t* tup_from_ssl_ctx(void *ssl_ctx, u64 pid_tgid) {
+    ssl_sock_t *ssl_sock = bpf_map_lookup_elem(&ssl_sock_by_ctx, &ssl_ctx);
+    if (ssl_sock == NULL) {
+        return NULL;
+    }
+
+    if (ssl_sock->tup.metadata) {
```

Comment:
This requires the connection be TCP or IPv6. While that is likely true almost all of the time, would it be simpler to check `sport` or `dport`, which must non-zero to be valid?

Reply:
humm, good point. I'll update it